lib/Agrammon/Web/Routes.pm6

```diff
@@ -60,6 +64,30 @@ sub static-content($root) is export {
     }
 }
 
+sub api-routes (Str $schema, $ws) {
+    openapi $schema.IO, {
+        # working
+        operation 'renameDataset', -> LoggedIn $user {
+            request-body -> ( :oldName($old-name), :newName($new-name) ) {
+                $ws.rename-dataset($user, $old-name, $new-name);
+                CATCH {
+                    note "$_";
+                    when X::Agrammon::DB::Dataset::AlreadyExists {
+                        conflict 'application/json', %(
+                            error => .message
+                        );
+                    }
+                    when X::Agrammon::DB::Dataset::RenameFailed {
+                        conflict 'application/json', %(
+                            error => .message
+                        );
+                    }
```

Comment:
Could be shortened to `when X::Agrammon::DB::Dataset::AlreadyExists | X::Agrammon::DB::Dataset::RenameFailed { ... }` given then `when` bodies are the same.

Reply:
Didn't know how to do that properly, thanks.

Reply:
Fixed